source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,276 @@
+snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/textSearch.go
+   :language: go
+   :dedent:
+   :start-after: begin insert docs
+   :end-before: end insert docs
+
+.. include:: /includes/fundamentals/automatic-db-coll-creation.rst
+
+Each document contains the name and release year of the Marvel movie
+that corresponds to the ``title`` and ``year`` fields.
+
+.. include:: /includes/fundamentals/truncated-id.rst
+
+Text Index
+~~~~~~~~~~
+
+You must create a **text index** before running a text search. A text
+index specifies the string or string array field to run a text
+search on.
+
+The examples in the following sections run text searches on the
+``title`` field in the ``movies`` collection. To enable text searches on
+the ``title`` field, create a text index using with the following snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/textSearch.go
+   :language: go
+   :dedent:
+   :start-after: begin text index
+   :end-before: end text index
+
+.. _text-search-go:
+
+Text Search
+-----------
+
+A text search retrieves documents that contain a **term** or a
+**phrase** in the text indexed fields. A term is a sequence of
+characters that excludes whitespace characters. A phrase is a sequence
+of terms with any number of whitespace characters.
+
+To perform a text search, use the ``$text`` evaluation query operator,
+followed by the ``$search`` field in your query filter. The ``$text`` operator
+performs a text search on the text indexed fields. The ``$search`` field
+specifies the text to search in the text indexed fields.
+
+Query filters for text searches use the following format:
+
+.. code-block:: go
+
+   filter := bson.D{{"$text", bson.D{{"$search", "<text to search>"}}}}
+
+.. _term-search-go:
+
+Search by a Term
+~~~~~~~~~~~~~~~~
+
+To search for a term, specify the term as a string in your query filter.
+To search for multiple terms, separate each term with spaces in the string.
+
+.. note::
+
+   When searching for multiple terms, the ``Find()`` function returns
+   documents with at least one of the terms in text indexed fields.
+
+Example
+```````
+
+The following example runs a text search for titles that contain the term "War":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/textSearch.go
+   :language: go
+   :dedent:
+   :start-after: begin term search
+   :end-before: end term search
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {title Avengers: Infinity War} {year 2018}]
+   [{_id ObjectID("...")} {title Captain America: Civil War} {year 2016}]
+
+Search by a Phrase
+~~~~~~~~~~~~~~~~~~
+
+To search for a phrase, specify the phrase with escaped quotes as a
+string in your query filter. If you don't add escaped quotes around the
+phrase, the ``Find()`` function runs a :ref:`term search <term-search-go>`.
```

Comment:
Ah... so, if term search allows multiple terms separated by spaces within a single string to search for individual terms, please ignore my above comment.